src/server/accept.rs

```diff
@@ -98,25 +98,12 @@ where
             // The server is at capacity. Try to acquire a permit with the
             // configured timeout.
             Err(_) => {
-                let acquire = semaphore.clone().acquire_owned();
-
-                tokio::select! {
-                    result = time::timeout(config.accept_timeout, acquire) => {
-                        match result {
-                            // Permit acquired!
-                            Ok(Ok(acquired)) => acquired,
-
-                            // The semaphore was dropped. Likely unreachable.
-                            Ok(Err(_)) => break ExitCode::FAILURE,
-
-                            // The server is still at capacity. Reset the connection.
-                            Err(_) => continue,
-                        }
-                    }
-
-                    _ = shutdown_rx.changed() => {
-                        break receive_ctrl_c!(shutdown_rx);
-                    }
+                task::yield_now().await;
```

Comment:
The cost of yielding here is likely <10 µs for most applications.

We are betting that a scheduler tick is all we need to stabilize a saturated cluster.

Any more and it wouldn't be worth it. Any less is no better than the status quo.